package com.anthropic.models.beta.messages

import com.anthropic.core.jsonMapper
import com.fasterxml.jackson.module.kotlin.jacksonTypeRef
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class BetaUrlImageSourceTest {

    @Test
    fun create() {
        val betaUrlImageSource = BetaUrlImageSource.of("url")

        assertThat(betaUrlImageSource.url()).isEqualTo("url")
    }

    @Test
    fun roundtrip() {
        val jsonMapper = jsonMapper()
        val betaUrlImageSource = BetaUrlImageSource.of("url")

        val roundtrippedBetaUrlImageSource =
            jsonMapper.readValue(
                jsonMapper.writeValueAsString(betaUrlImageSource),
                jacksonTypeRef<BetaUrlImageSource>(),
            )

        assertThat(roundtrippedBetaUrlImageSource).isEqualTo(betaUrlImageSource)
    }
}
